select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
+
+        // Immediately invoke any methods needed to set up initial state.
+        this.update();
+    }
+
+    //
+    // Update the component’s visible value when the underlying select element’s
+    // value changes.
+    //
+    Select.prototype.update = function update() {
+        this.value = this.dom.$root.find('option:selected').text();
+        this.dom.$value.text(this.value);
+        this.dom.$root.trigger('select:update'); // notify subscribers
```

Comment:
Oh, yeah, you’re right. Since there is no base component JS to inherit from, any idea how to do this? I don’t think we want to maintain the implementation in each component’s script. A mixin would be a good option if we don’t want to do actual inheritance. Thoughts?